src/Symfony/Component/Security/Core/User/UserChecker.php

```diff
@@ -16,10 +16,14 @@
 use Symfony\Component\Security\Core\Exception\DisabledException;
 use Symfony\Component\Security\Core\Exception\LockedException;
 
+trigger_deprecation('symfony/security-core', '5.3', 'The "%s" class is deprecated, use "%s" instead.', UserChecker::class, InMemoryUserChecker::class);
+
 /**
  * UserChecker checks the user account flags.
  *
  * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @deprecated since Symfony 5.3, use {@link InMemoryUserChecker} instead
```

Comment:
Shouldn't we instead use `class_alias`, like here:
https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/Messenger/Transport/AmqpExt/AmqpFactory.php#L16
https://github.com/symfony/symfony/blob/5.x/src/Symfony/Component/Messenger/Bridge/Amqp/Transport/AmqpFactory.php#L36